### PR TITLE
fix key error from log command

### LINKED ISSUE #19: API changes of the new central service
Tested at CSX, observed the following issue:
```py
In [2]: from pyOlog import SimpleOlogClient

In [3]: soc = SimpleOlogClient()

In [7]: soc.session._url
Out[7]: 'https://<REDACTED>/Olog'

In [10]: soc.log("Test of https://<REDACTED>/Olog by @mrakitin")
/opt/conda_envs/nsls2-collection-2021-2.2/lib/python3.7/site-packages/urllib3/connectionpool.py:1020: InsecureRequestWarning: Unverified HTTPS request is being made to host '<REDACTED>'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-10-763d8de3e4cf> in <module>
----> 1 soc.log("Test of https://<REDACTED>/Olog by @mrakitin")

/opt/conda_envs/nsls2-collection-2021-2.2/lib/python3.7/site-packages/pyOlog/SimpleOlogClient.py in log(self, text, logbooks, tags, properties, attachments, verify, ensure)
    263         log_entry = self._build_entry(text, logbooks, tags, properties,
    264                                       attachments, verify, ensure)
--> 265         return self.session.log(log_entry)
    266 
    267     def update(self, log_id, text=None, logbooks=None, tags=None,

/opt/conda_envs/nsls2-collection-2021-2.2/lib/python3.7/site-packages/pyOlog/OlogClient.py in log(self, log_entry)
    131         resp = self._post(self.logs_resource,
    132                           data=LogEntryEncoder().encode(log_entry))
--> 133         id = LogEntryDecoder().dictToLogEntry(resp.json()[0]).id
    134 
    135         # Handle attachments

KeyError: 0

In [11]: %debug
> /opt/conda_envs/nsls2-collection-2021-2.2/lib/python3.7/site-packages/pyOlog/OlogClient.py(133)log()
    131         resp = self._post(self.logs_resource,
    132                           data=LogEntryEncoder().encode(log_entry))
--> 133         id = LogEntryDecoder().dictToLogEntry(resp.json()[0]).id
    134 
    135         # Handle attachments

ipdb> p resp
<Response [200]>

ipdb> p resp.json()
{'count': None, 'log': [{'createdDate': 1640272672541, 'modifiedDate': 1640272672542, 'owner': '<REDACTED>', 'source': '<REDACTED>', 'version': '1', 'description': 'Test of https://<REDACTED>/Olog by @mrakitin', 'logbooks': [{'owner': 'olog-logs', 'id': 1, 'name': 'Commissioning', 'state': 'Active'}], 'tags': [], 'properties': [], 'attachments': [], 'id': 131483, 'level': 'Info', 'state': 'Active', 'eventStart': 1640272672530, 'eventEnd': 38742530400000}]}

ipdb> p LogEntryDecoder().dictToLogEntry(resp.json()['log'][0]).id
131483
```

### PR DESCRIPTION
**Pull request by gwbischof**
Fixes #19

### CODE REVIEW

File: pyOlog/OlogClient.py [hunk 1]
```diff
@@ -60,6 +60,9 @@ def __init__(self, url=None, username=None, password=None, ask=True):
         self.verify = False
         username = _conf.get_username(username)
         password = _conf.get_value('password', password)
+        self._old_olog_api = _conf.get_value('old_olog_api', old_olog_api)
+        if self._old_olog_api == 'true' or self._old_olog_api == 'True`':
```
Comment (mrakitin) [inline pyOlog/OlogClient.py]:
```suggestion
        if self._old_olog_api is not None and self._old_olog_api.lower() == 'true':
```

File: pyOlog/OlogClient.py [hunk 2]
```diff
@@ -60,6 +60,9 @@ def __init__(self, url=None, username=None, password=None, ask=True):
         self.verify = False
         username = _conf.get_username(username)
         password = _conf.get_value('password', password)
+        self._old_olog_api = _conf.get_value('old_olog_api', old_olog_api)
```
Comment (mrakitin) [inline pyOlog/OlogClient.py:67]: Will it handle the case when the value is not set in the config file?

**Review by mrakitin — COMMENTED** [review]
Looks good to me. Have just a single question about the implementation.

Also, could you please also provide a sample `.pyOlog.conf` file (without real user name/password)? That will help as a future reference.

File: pyOlog/OlogClient.py (hunk 2 above)
Comment (gwbischof) [reply pyOlog/OlogClient.py:67]: Yep, there is some logic that does that.
I updated and tested this code more.

**Comment by gwbischof** [comment]
Without `old_olog_api` defined in the config file.
```
In [3]: from pyOlog import SimpleOlogClient

In [4]: soc = SimpleOlogClient(old_olog_api=True)
True

In [5]: soc = SimpleOlogClient(old_olog_api=False)
False

In [6]: soc = SimpleOlogClient()
False
```

With `old_olog_api` defined, but not equal to `true` or `True`
```
In [1]: from pyOlog import SimpleOlogClient

In [2]: soc = SimpleOlogClient()
False
```

With `old_olog_api` set to `true`
```
In [1]: from pyOlog import SimpleOlogClient

In [2]: soc = SimpleOlogClient()
True
```

Example config file:
```
[DEFAULT]
url=http://epics-services-tst:7070/tst_logbook
username=user
password=pass
logbooks=Commissioning
default_owner=olog_logs
old_olog_api=true
```

File: pyOlog/OlogClient.py [hunk 3]
```diff
@@ -44,22 +44,31 @@ class OlogClient(object):
...
         '''
         self._url = _conf.get_value('url', url)
         self.verify = False
         username = _conf.get_username(username)
         password = _conf.get_value('password', password)
+        self._old_olog_api = _conf.get_value('old_olog_api', old_olog_api)
+        self._old_olog_api = (self._old_olog_api == True 
+                              or self._old_olog_api == 'True' 
+                              or self._old_olog_api == 'true')
+        print(self._old_olog_api)
```
Comment (mrakitin) [inline pyOlog/OlogClient.py]: I suggest to remove this print from from the production code:
```suggestion
```

**Review by mrakitin — APPROVED** [review]
Looks good to me, thanks for addressing the issue!

I am suggesting removing the print from the production code, as it can be accessed via a private attribute if needed.

**Pull request merged**